src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php

```diff
@@ -460,14 +475,86 @@ protected function detachUsingCustomClass($ids)
         return $results;
     }
 
+    /**
+     * Restore the intermediate table entries with a list of IDs or collection of models.
+     *
+     * @param  mixed  $ids
+     * @param  bool  $touch
+     * @return int
+     */
+    public function restore($ids = null, $touch = true)
+    {
+        if ($this->using && ! empty($ids) && empty($this->pivotWheres) && empty($this->pivotWhereIns)) {
+            $results = $this->restoreUsingCustomClass($ids);
+        } else {
+            $query = $this->newPivotQuery();
+
+            // If associated IDs were passed to the method we will only restore those
+            // associations, otherwise all of the association ties will be broken.
+            // We'll return the numbers of affected rows when we do the restores.
+            if (! is_null($ids)) {
+                $ids = $this->parseIds($ids);
+
+                if (empty($ids)) {
+                    return 0;
+                }
+
+                $query->whereIn($this->relatedPivotKey, (array) $ids);
+            }
+
+            // Once we have all of the conditions set on the statement, we are ready
+            // to run the restore on the pivot table. Then, if the touch parameter
+            // is true, we will go ahead and touch all related models to sync.
+            if ($this->withSoftDeletes) {
+                $attributes = [
+                    $this->deletedAt() => null,
+                ];
+
+                if ($this->hasPivotColumn($this->updatedAt())) {
+                    $attributes[$this->updatedAt()] = now();
+                }
+
+                $results = $query->update($attributes);
+            } else {
+                $results = $query->delete();
```

Comment:
I don't understand this line. Why would the `restore` method delete anything?

Reply:
thank you, fixed it